### ISSUE TITLE
Linear Regression

### ISSUE THREAD

**Issue by oke-aditya** [madewithml/hacktoberfest#13]
## 🚀 Tutorial / New Example Request

The most underrated topic in ML Linear Regression.

**Describe the solution you'd like**

1. Walk in-depth guide to linear regression. The problem formulation to how we solve it using the Matrix method. As well as gradient descent.
2. There is a lot to explore with regularization in linear regression. Especially Ridge, Lasso and other regressions.
3. A lot of content is available in sklearn documentation (please don't' simply copy-paste, we will check that) Also book by Hastie and Friedman (Elements of statistical learning has great content)
4. Formulate the idea into simple bi-variate regression (since It will be easy to visualize) generalize it on common dataset available in sklearn.

Keep this intuitive and super beginner-friendly, coz this is probably first topic people will learn from, so it has to be really good.

**Describe alternatives you've considered**
I think we should keep this simple there are other linear methods but we can cover them later.

**Comment by premalrupnur** [comment]
Hey may I would like to go with this issue, may I??

**Comment by premalrupnur** [comment]
I had a doubt , if we had to go with bi variate regression, according to me we can't explain the regularization techniques intuitively with it , also what I think is polynomial regression can be used as simple example to give better insights on regularization techniques

**Comment by premalrupnur** [comment]
Do let me know your views on it

**Comment by oke-aditya** [comment]
> 
> 
> I had a doubt , if we had to go with bi variate regression, according to me we can't explain the regularization techniques intuitively with it .

I don't think so, I certainly think it is possible and even more effective with 2 variables or 1 variable case.

You can work on this issue, but ensure that you understand what exactly needs to be in.

It would be easier to propose a brief layout of your Jupyter Notebook and then start working.

**Comment by premalrupnur** [comment]
Yes I would create a pr with the brief layout of notebook

**Comment by premalrupnur** [comment]
> > I had a doubt , if we had to go with bi variate regression, according to me we can't explain the regularization techniques intuitively with it .
> 
> I don't think so, I certainly think it is possible and even more effective with 2 variables or 1 variable case.
> 
ok I would use bi-variate to explain regularization techniques

**Comment by premalrupnur** [comment]
Do I have to import linear_model from  sklearn or implement everything from scratch using numpy??

**Comment by oke-aditya** [comment]
From `sklearn`. I think scratch is bit overkill and too much for new people.

**Comment by premalrupnur** [comment]
Is it mandatory to use sklearn dataset what if we use array of numbers and fit to it the linear_model because sklearn datasets are multifeatured and would require preprocessing

**Comment by oke-aditya** [comment]
There are functions such as `dataset.make_regression` which make it easy to create dummy datasets for regression.

You will get a good fitting curve as well.

**Comment by premalrupnur** [comment]
If we aren't scratch training how can gradient descent be explained

**Comment by premalrupnur** [comment]
May I use  some images and explanation's from Andrew Ng's  ML course those images are really intuitive for understanding cost function

**Comment by oke-aditya** [comment]
> If we aren't scratch training how can gradient descent be explained also matrix manipulations would also be skipped 
> 

You can explain them. I think it's hard to understand implementation for beginners to These. 
(It's very simple with autograd based libraries like torch)

You can see Sklearn documentation and API and explain which method it does too. And how gradient descent might be better than matrix method it vice versa.

**Comment by oke-aditya** [comment]
> May I use  some images and explanation's from Andrew Ng's  ML course those images are really intuitive for understanding cost function

I think you should cite it properly if you use them. 

Guess this question is common for images / content. 

@GokuMohandas what should be best for taking such content. I will add this to contributing.md file too.

**Comment by GokuMohandas** [comment]
If they have a generated BibTeX citation then we should use that (you can paste them at the bottom under a **References** section). However, if they don't have one readily available, you can either generate a BibTeX citation yourself like this:

```
@misc{{org name},
  title = {}, 
  year = {},
  note = {},
  url={},
  author = {},
}
```
or at the very least, you can reference the link directly under the resource text, image, etc.

When you use the BibTeX citation, you can refer to the resources by a number in the text and then in the **References** section, each number has an associated BibTeX citation.

**Comment by premalrupnur** [comment]
> I don't think so, I certainly think it is possible and even more effective with 2 variables or 1 variable case.

yes right actually I meant using bivariate polynomial regression for explaining regularization , just got confused between bivariate and polynomial regression yesterday😄

**Comment by premalrupnur** [comment]
Hi I think I have not explained how regularisation prevents overfitting  properly,  what are your views on it
